designsafe/apps/api/projects_v2/management/commands/migrate_projects_tapis_systems_from_v2_to_v3.py

```diff
@@ -0,0 +1,244 @@
+        "id": system_id,
+        "host": HOST,
+        "port": int(PORT),
+        "systemType": "LINUX",
+        "defaultAuthnMethod": "PKI_KEYS",
+        "canExec": False,
+        "rootDir": project_root_dir,
+        "effectiveUserId": TG_USER,
+        "authnCredential": {
+            "privateKey": TG_USER_PRIVATE_KEY,
+            "publicKey": TG_USER_PUBLIC_KEY,
+        },
+        "notes": {"title": title, "description": description},
+    }
+    if owner:
+        system_args["owner"] = owner
+    if create:
+        client.systems.createSystem(**system_args)
+    else:
+        client.systems.patchSystem(systemId=system_id, **system_args)
+
+
+# pylint: disable=invalid-name
+HOST = settings.PROJECT_STORAGE_SYSTEM_TEMPLATE["storage"][
+    "host"
+]  # cloud.corral.tacc.utexas.edu
+PORT = 22
+TG_USER = settings.PROJECT_STORAGE_SYSTEM_TEMPLATE["storage"]["auth"][
+    "username"
+]  # i.e. tg458981
+TG_USER_PRIVATE_KEY = settings.PROJECT_STORAGE_SYSTEM_TEMPLATE["storage"]["auth"][
+    "privateKey"
+]
+TG_USER_PUBLIC_KEY = settings.PROJECT_STORAGE_SYSTEM_TEMPLATE["storage"]["auth"][
+    "publicKey"
+]
+OWNER = "wma_prtl"
+# pylint: enable=invalid-name
+
+
+class Command(BaseCommand):
+    """Command for migrating projects from Tapis v2 to v3"""
+
+    help = (
+        "Facilitates the migration of project systems from Tapis v2 to v3,"
+        " with options for a dry run and updating existing systems."
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Executes the command in a simulation mode, logging actions "
+            "without applying any changes to the systems or data.",
+        )
+
+        parser.add_argument(
+            "--update-existing",
+            action="store_true",
+            help="Allows the command to update systems that already exist in Tapis V3,"
+            "ensuring they are synchronized with their V2 counterparts.",
+        )
+
+    def handle(
+        self, *args, **options
+    ):  # pylint: disable=too-many-statements disable=too-many-locals
+        dry_run = options["dry_run"]
+        update_existing = options["update_existing"]
+
+        TAPIS_TENANT_BASEURL = os.environ.get(  # pylint: disable=invalid-name
+            "TAPIS_TENANT_BASEURL"
+        )
+        TAPIS_ADMIN_JWT = os.environ.get(  # pylint: disable=invalid-name
+            "TAPIS_ADMIN_JWT"
+        )
+        client = Tapis(base_url=TAPIS_TENANT_BASEURL, access_token=TAPIS_ADMIN_JWT)
+
+        if dry_run:
+            logger.info("Running in dry-run mode. No changes will be made.")
+
+        total_number_projects = len(list(iterate_entities()))
+
+        for i, project in enumerate(iterate_entities()):
+            uuid = project["uuid"]
+            project_id = project["value"]["projectId"]
+            system = f"project-{uuid}"
+            project_root_dir = f"/corral-repl/projects/NHERI/projects/{uuid}"
```

Comment:
@jarosenb .  adding note from slack: we need custom handling for PRJ-1305 which is the special project used to manage data for community data needs to be handled.